src/multihash.rs

```diff
@@ -259,6 +275,37 @@ where
     Ok(())
 }
 
+/// Writes the multihash to a byte slice.
+#[cfg(not(feature = "std"))]
+pub fn write_multihash<'a>(
+    w: &'a mut &'a mut [u8],
+    code: u64,
+    size: u8,
+    digest: &[u8],
+) -> Result<(), Error> {
+    use unsigned_varint::encode as varint_encode;
+
+    let mut code_buf = varint_encode::u64_buffer();
+    let code = varint_encode::u64(code, &mut code_buf);
+
+    let mut size_buf = varint_encode::u8_buffer();
+    let size = varint_encode::u8(size, &mut size_buf);
```

Comment:
This part could be re-used for both, the std and non-std implementation. I'm not sure if it makes the code easier to read, but it might be worth a try.